### PR TITLE
Added some warning text about fractional IRs to IR selection description

### PR DESCRIPTION
**Pull request by aserna3**
~~I closed my last PR: https://github.com/dolphin-emu/dolphin/pull/2734 because this is a different change and there were some issues I was having with syncing my fork, rebasing, and file conflicts from code I was trying to fix (Which fixed what I was intending on fixing)~~

~~Anyways, I feel that fractional IRs should be removed from the GUI. For the end user they can cause a lot of headaches and the only users that should use them, probably know why (iGPU users). These users will still have access to all fractional IRs through GameINI with this change, so I'm not removing any features.~~

I yield. I added some warning text instead. yay.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
I haven't actually tested it, but I suspect the GUI dropdown no longer maps the values correctly to iEFBScale with this. Before the values corresponded 1:1 to the [EFBScale enum](https://github.com/dolphin-emu/dolphin/blob/8493feb89f86d395cd5680a464cf296e26ab9f29/Source/Core/VideoCommon/VideoConfig.h#L34), now they don't, which makes me believe that selecting the "Native" option would actually internally set SCALE_AUTO_INTEGRAL, selecting "2x Native" would actually set SCALE_1X etc.

**Comment by aserna3** [comment]
I was just made aware of this. I don't know how to fix it without removing those enums which every time I've touched those, broke everything. Any ideas?

**Comment by AdmiralCurtiss** [comment]
I fear that you'll just have to fix it! Or let someone else do it I guess.

**Comment by AdmiralCurtiss** [comment]
Here's some tips I see looking through this:
* In RenderBase::CalculateTargetSize(), remove all the cases except default and SCALE_AUTO_INTEGRAL, and adjust the s_last_efb_scale - 3 in the default case. (presumably to just s_last_efb_scale)
* RenderBase::DrawDebugText(), just remove the extra cases.
* That block in VideoConfig::GameIniLoad() needs to be changed somehow, I guess it would be enough to just remove the else case?
* In RenderBase there's a few places where you can just remove the SCALE_AUTO case.
* And in Frame::ParseHotkeys you change the SCALE_AUTO to SCALE_AUTO_INTEGRAL.

**Comment by BlackTelomeres** [comment]
You do realize that 95% of iGPU users will not know that they can still do this through INIs and thus you are functionally removing the ability for them to use 1.5 IR, right?

**Comment by comex** [comment]
What is wrong with fractional IRs?

**Comment by aserna3** [comment]
There are games that don't like it at all. Off the top of my head, skyward sword in OpenGL is a huge one this breaks. https://code.google.com/p/dolphin-emu/issues/detail?id=6640

I'm sure there's others if I looked through the issue tracker / forums.

**Comment by comex** [comment]
Meh.  I think all IRs other than 1 are a hack, since they make it impossible to do the right thing if the CPU touches the framebuffer directly; I'm not convinced the situation with non-integral multipliers is so much worse that issues can't be worked around.

But feel free to enlighten me, graphics folks...

**Comment by BlackTelomeres** [comment]
So is the main issue then that you can break the graphics in some games like SS by selecting an incompatible option in a drop down? But it's okay if you can edit an option in an ini file to break the game? Or do you want to make it less likely they will break the graphics in one of these games by making it less likely they will use a fractional IR since it would no longer be in the GUI?

**Comment by i30817** [comment]
I'm probably one of the persons with a shitter computer playing dolphin ATM and i wouldn't DREAM of using it with anything else than native resolution with any effects whatsoever and probably not in fullscreen.
Resuming some people complaining about being forced to choose 2.0 or 1.0 instead of 1.5 doth complain too much.

File: Source/Core/VideoCommon/VideoConfig.h [hunk 1]
```diff
@@ -34,12 +34,9 @@ enum AspectMode
 enum EFBScale
 {
 	SCALE_FORCE_INTEGRAL = -1,
```
Comment (degasus) [inline Source/Core/VideoCommon/VideoConfig.h]: There is no need for "SCALE_FORCE_INTEGRAL" anymore at all.

File: Source/Core/VideoCommon/RenderBase.cpp [hunk 2]
```diff
@@ -331,26 +296,14 @@ void Renderer::DrawDebugText()
 		std::string res_text;
 		switch (g_ActiveConfig.iEFBScale)
 		{
-		case SCALE_AUTO:
-			res_text = "Auto (fractional)";
-			break;
 		case SCALE_AUTO_INTEGRAL:
 			res_text = "Auto (integral)";
```
Comment (degasus) [inline Source/Core/VideoCommon/RenderBase.cpp]: If we won't support fractional scaling avy more, than there is no need to be explicit here:
res_text = "Auto";

Also, I think we should rename the enum SCALE_AUTO_INTEGRAL to SCALE_AUTO.

File: Source/Core/DolphinWX/VideoConfigDiag.cpp [hunk 3]
```diff
@@ -347,18 +347,17 @@ VideoConfigDiag::VideoConfigDiag(wxWindow* parent, const std::string &title, con
...
 	{
-	const wxString efbscale_choices[] = { _("Auto (Window Size)"), _("Auto (Multiple of 640x528)"),
-		_("Native (640x528)"), _("1.5x Native (960x792)"), _("2x Native (1280x1056) for 720p"), _("2.5x Native (1600x1320)"),
+	const wxString efbscale_choices[] = { _("Auto (Multiple of 640x528)"), _("Native (640x528)"), _("2x Native (1280x1056) for 720p"),
 		_("3x Native (1920x1584) for 1080p"), _("4x Native (2560x2112) for 1440p"), _("5x Native (3200x2640)"),
 		_("6x Native (3840x3168) for 4K"), _("7x Native (4480x3696)"), _("8x Native (5120x4224) for 5K"), _("Custom") };
 
 	wxChoice *const choice_efbscale = CreateChoice(page_enh,
-		vconfig.iEFBScale, wxGetTranslation(internal_res_desc), (vconfig.iEFBScale > 11) ?
+		vconfig.iEFBScale, wxGetTranslation(internal_res_desc), (vconfig.iEFBScale > SCALE_MAX_GUI) ?
```
Comment (degasus) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: Do you need SCALE_MAX_GUI at all?

const int SCALE_MAX_GUI = sizeof(efbscale_choices) / sizeof(efbscale_choices[0]);

File: Source/Core/VideoCommon/VideoConfig.cpp [hunk 4]
```diff
@@ -168,23 +168,6 @@ void VideoConfig::GameIniLoad()
 		{
 			iEFBScale = tmp;
 		}
-		else // Round down to multiple of native IR
```
Comment (degasus) [inline Source/Core/VideoCommon/VideoConfig.cpp]: I think it should be fine to read iEFBScale directly:
CHECK_SETTING("Video_Settings", "EFBScale", iEFBScale );
Maybe you also need to touch all affected game inis :/

**Comment by aserna3** [comment]
Okay, so about the games that break on fractional IRs. Phire did a grep against our gameini set and found quite a few here: https://gist.github.com/phire/169ab7d43d066f8317f6 Quite a lot. Some of them popular. I also believe it breaks SSAA in many games.

And that's only that we know of. Phire said there's likely more that we're just not aware of yet. Anyways.  As much as I love our iGPU users, I still don't think we should encourage fractional IRs. I view having something in the GUI to say "We tested this thing. It works fine." when, well, fractional IRs are a pretty big hack. Thats why I think it's a feature that should be kept to gameinis, so only people that know that they're probably going to break stuff will use it.

**Comment by Linktothepast** [comment]
There are also a lot of games that use EFBScale = 2 that is the equivalent of 1x, mainly virtual console games (nes etc.). If this change is merged those gameinis will need to be altered probably since it won't be the number 2 drop down option any more when you are counting from 0.

**Comment by tony971** [comment]
This might be a silly question, but why not just create a new ini setting for games that have issues with fractional IRs? Similar to EFBScale, but just rounding down to the nearest integer. That way no one complains and games known to have issues don't break. Because like @comex said, all non-native IRs are a hack anyway. If @phire already compiled a list of affected games, then half the legwork is already done.

**Comment by aserna3** [comment]
Those are the games that we're aware of, and regressions / fixes can happen, which modify that list.

**Comment by BhaaLseN** [comment]
My initial suggestion was to reorder the enum values to make sense afterwards, since we'll end up breaking things one way or the other (at least the fractional guys, but also the force_whatever ones and the overpowered people that selected 3x or higher).
And even tho fixing the existing game INIs that force Native res might be a lot of work; it is a once-only task. We _could_ compromise to start counting at 2 for Native (if someone - @Linktothepast - objects), but that would make the other settings a bit awkward ("what, 4x IR has the value 5?") and has the potential of being confusing in the future (then again; when will we ever need to change that enum again? everything is possible from INI, and its mostly only there for the GUI at this point).
I'd go for breaking it completely, not just a bit :)

**Comment by TurboK234** [comment]
Is having useful (for me, as Intel GPU user) but less-than-100% compatible options (meaning 1,5x and 2,5x) in Graphics/Enhancements really _that_ bad that just adding a line in the Description field isn't enough? (like: "Using a fractional scaling factor like 1.5x and 2.5x might cause glitches and using them is not recommended").
I mean, AA, AF, Fog, Force Filtering, PPL and Widescreen Hack are all known to be less-than-100% compatible, or have been at some point.
But I understand if the GUI lines have to go, as long as there is the INI option.

**Comment by Linktothepast** [comment]
@BhaaLseN For mass editing i am using a program called TextCrawler for a while now, it is not that difficult to do so therefore don't mind me i will manage to keep the gameini database up to date and i prefer proper clean solutions without jumping numbers or anything. I just write comments to remember it when the pr gets merged not to nag about it or anything.

**Comment by comex** [comment]
In the long term I think there should be care taken to fix or work around such glitches wherever reasonably possible... just personally.

But in the short term, it's silly to hide an option in an INI file 99% of people will have no idea how to find, *if* some users significantly benefit from it.  (After all, the feature still exists, so hiding it doesn't save any codebase complexity.)   Better to warn in the GUI.

**Comment by BlackTelomeres** [comment]
>I view having something in the GUI to say "We tested this thing. It works fine."

You should probably remove most of the options under Graphics>Hacks because I assure you if used on the wrong games they can break things, just like fractional IRs.

**Comment by Rukario** [comment]
Did someone just said Graphics>Hacks? If we want to remove fractional IR from dropdown menu FROM Graphics>Enhancement because it's more of a >HACK< and can break some games (more so than integral IR's)... Hint hint.

**Comment by i30817** [comment]
I think that would be too complicated a procedure (and confusing to users to boot). If they really really want to break their graphics they shouldn't be scared of a config file.

This is just my personal opinon, thou. I'm a nobody here.

**Comment by BlackTelomeres** [comment]
Well some people have put forth the idea that increasing internal resolution at all is basically a hack. But probably everything under Enhancements qualifies as a hack anyway, so people should expect them to potentially cause issues in some games.

**Comment by yorkiesandskittles** [comment]
I still think that having a "custom" resolution option like PCSX2 would be
great. This means we can have fractional IRs and absolutely insane
resolutions under one option (we should also have an 'Apply' button on the
Enhancements tab to avoid people possibly accidentally inputting one number
too many and ending up with a resolution that could slaughter the VRAM or
even cause the system to hang altogether.)

**Comment by cammelspit** [comment]
+1 for easy user configurable IRs

**Comment by i30817** [comment]
Great, so a push to remove bugs will multiply them when people want to support custom resolution 10.9532.

**Comment by Husono** [comment]
@yorkiesandskittles That would just amplify the problem this pr is trying to address. While anything that is not 1x IR is a hack, fractional IRs lead to more problems than whole number IRs. It is because of this that they are being dropped from the gui. 

Reasons for dropping each option from the gui: 
Auto (Window Size) should be dropped because fractional IRs are not recommended for an automatic setting. 
2.5x IR should be dropped because it does offer that much of an improvement over 2x and no one has spoken up for it so far. 
1.5x IR is still commonly used by integrated graphics users and provides more significant improvement over 2.5x IR. However it is still prone to the issues of fractional IR and therefore should not be in the gui. If not for this option I would recommend removing fractional IR from Dolphin entirely. 

@degasus With the current pr would users still be able to set 1.5x IR in the ini files?

**Comment by aserna3** [comment]
Yes, I tested with a couple games. setting IR in gameini still works.

**Comment by AdmiralCurtiss** [comment]
If I may ask, how? The meaning of the EFBScale ini settings is definitely modified with this.

**Comment by aserna3** [comment]
*double checks*

... And it's not working anymore. I swore I tested this when I made the PR. I was positive it was working.

Looks like I'll need to figure out someway to separate fractional IRs into a different setting...

**Comment by degasus** [comment]
How shall it work? You've removed the code which handles fractional IRs ;)

**Comment by aserna3** [comment]
I yield. The iGPU defense force makes good points and I'm rolling everything back. I'm just going to put a text warning in the IR description field and call it a day. Obviously I'll edit my PR.

**Comment by lioncash** [comment]
Looks fine to me.

**Comment by Rukario** [comment]
Come on, you could remove fractional IRs from dropdown menu and implement checkbox in Graphics>Hacks.

"Add 0.5x to IR Multiplier" with checkbox before it. Then write long sentences about why does it exist and why is it bad in the description.

**Comment by aserna3** [comment]
It;s surprisingly difficult to do that without removing fractional IRs entirely. the combobox is coded so that it's 1:1 with the scale enums. Removing any of the fractional settings breaks stuff. To fix what it breaks, you have to mess around with the enums, which then breaks gameini and inc/dec IR keys, and to fix that you have to remove fractional IRs entirely. Everything about fractional IRs is tied very closely to the EFBScale enum.

**Comment by Rukario** [comment]
Then it will be a greater accomplishment. But it's fine if you can't do it.

**Comment by JosJuice** [comment]
Splitting the resolution setting into one setting in the Enhancements tab and one setting on the Hacks tab would just be confusing, as users would need to be aware of both tabs to know which resolution they are using. I'm in favor of the current version of this PR.

**Comment by Rukario** [comment]
It will stop people from not reading (description) and more people will overlook that setting in Hacks tab. Isn't that a good thing?

There is one free space for one more checkbox in Enhancement tab as well. Either that or in Hacks tab.

**Comment by lioncash** [comment]
> It will stop people from not reading (description) and more people will overlook that setting in Hacks tab. Isn't that a good thing?

That's just terrible UX in general.

**Comment by aserna3** [comment]
What's the status on this?

**Pull request merged**